core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/HttpWorkspaceManagerDAO.scala

```diff
@@ -201,4 +201,20 @@ class HttpWorkspaceManagerDAO(apiClientProvider: WorkspaceManagerApiClientProvid
         ),
       workspaceId
     )
+
+  def cloneSnapshotByReference(sourceWorkspaceId: UUID,
+                               snapshotId: UUID,
+                               destinationWorkspaceId: UUID,
+                               name: String,
+                               ctx: RawlsRequestContext
+  ): DataRepoSnapshotResource = {
+    val requestBody = new CloneReferencedResourceRequestBody()
```

Comment:
There's no real need for this to return `DataRepoSnapshotResource` as I don't think it's ever used, although in theory it could be under some future use cases.

Reply:
have you inspected the `DataRepoSnapshotResource` that is actually returned from this call? From a quick read of the code, I think it might always be null - though I could have misread